spring-kafka/src/main/java/org/springframework/kafka/listener/CommonDelegatingErrorHandler.java

```diff
@@ -173,4 +182,29 @@ private CommonErrorHandler findDelegate(Throwable thrownException) {
 		return null;
 	}
 
+	@Nullable
+	private Throwable findCause(Throwable thrownException) {
+		if (this.deepCauseChainTraversing) {
+			return deepTraverseCauseChain(thrownException);
+		}
+		return shallowTraverseCauseChain(thrownException);
+	}
+
+	@Nullable
+	private Throwable shallowTraverseCauseChain(Throwable thrownException) {
+		Throwable cause = thrownException;
+		if (cause instanceof ListenerExecutionFailedException) {
+			cause = thrownException.getCause();
+		}
+		return cause;
+	}
+
+	@Nullable
+	private Throwable deepTraverseCauseChain(Throwable thrownException) {
+		while (thrownException != null && thrownException.getCause() != null) {
+			thrownException = thrownException.getCause();
+		}
+		return thrownException;
+	}
```

Comment:
Hello,
Looks like only the "root" exception will be checked for matching after that. In my initial request, I'd been thinking about more complex solution - to check every exception from the chain. This approach is used e.g. in `spring-retry` library (see `RetryTemplateBuilder.traversingCauses()`).
Will it be possible to improve the current logic with my suggestion?

Reply:
Hi, I just pushed a commit with improvements, tried to follow idea you mentioned. Looking forward to hear whether solution satisfies your needs now

Reply:
Actually, don't know much about all of these classifiers, but for me it's ok if it works :) 
Also, you'd probably need to actualize the `setCauseChainTraversing` javadoc